Objects/odictobject.c

```diff
@@ -984,6 +985,7 @@ odict_reduce(register PyODictObject *od, PyObject *Py_UNUSED(ignored))
 
 
 /*[clinic input]
+@critical_section
```

Comment:
This looks like it will call functions that themselves re-acquire the critical section.